commit dae7f7802d61b104acf200dd3c3408fe54a71eb8
Author: Olly Betts <olly@survex.com>
Date:   Mon Apr 8 16:34:36 2024 +1200

    Fix reported found token in some reading errors
    
    This happened to work on at least x86-64, but failed on some architectures.

diff --git a/src/datain.c b/src/datain.c
index 598473c4..0fee0dcc 100644
--- a/src/datain.c
+++ b/src/datain.c
@@ -397,7 +397,7 @@ compile_diagnostic_token_show(int diag_flags, int en)
    }
    if (!s_empty(&p)) {
       caret_width = s_len(&p);
-      compile_diagnostic(diag_flags|DIAG_COL, en, p);
+      compile_diagnostic(diag_flags|DIAG_COL, en, s_str(&p));
       caret_width = 0;
       s_free(&p);
    } else {
